django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,14 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
             if pk:
+                # MySQLism: Columns in HAVING clause must be added to the GROUP BY.
                 expressions = [pk] + [expr for expr in expressions if expr in having]
+        elif self.connection.features.allows_group_by_selected_pks:
+            pks = {expr for expr in expressions if hasattr(expr, 'target') and expr.target.primary_key}
+            aliases = {expr.alias for expr in pks}
+            expressions = [
+                expr for expr in expressions if expr in pks or getattr(expr, 'alias', None) not in aliases
```

Comment:
We want a comment here what we are doing, and why we are doing it.

I'm wondering what kind of expressions that do not have alias end up in here. Everything should come through get_group_by_cols(), and those cols should be "raw columns", that is direct references to alias.col. Seems like that isn't the case. It might be possible to fix this in a way that only alias.col references end up in here in the first place.